src/open_clip/hf_model.py

```diff
@@ -95,22 +84,29 @@ def __init__(
         if transformers is None:
             raise RuntimeError("Please `pip install transformers` to use pre-trained HuggingFace models")
         if config is None:
-          self.config = AutoConfig.from_pretrained(model_name_or_path)
-          self.transformer = AutoModel.from_pretrained(model_name_or_path)
+            self.config = AutoConfig.from_pretrained(model_name_or_path)
+            self.transformer = AutoModel.from_pretrained(model_name_or_path)
         else:
-          self.config = config
-          self.transformer = AutoModel.from_config(config)
-        
-        self.pooler = get_pooler(pooler_type)
-        d_model = self.config.hidden_size # TODO: get d_model from config
-        # different models can have different names for it
-        # ?? do we use separate classes for different archs or handle it with helper funcs  
+            self.config = config
+            self.transformer = AutoModel.from_config(config)
+
+        if pooler_type is None: # get default arch pooler
+            self.pooler = _POOLERS[(arch_dict[self.config.model_type]["pooler"])]()
+        else:
+            self.pooler = _POOLERS[get_pooler(pooler_type)]()
```

Comment:
shouldn't this be `self.pooler = _POOLERS[pooler_type]()`?

Reply:
yup, removed.